lib/registrations/registration_checker.rb

```diff
@@ -2,20 +2,41 @@
 
 module Registrations
   class RegistrationChecker
+    def self.apply_payload(registration, raw_payload)
+      guests = raw_payload['guests']
+
+      registration.guests = guests.to_i if raw_payload.key?('guests')
+
+      competing_payload = raw_payload['competing']
+      comment = competing_payload&.dig('comment')
+      organizer_comment = competing_payload&.dig('organizer_comment')
+
+      registration.comments = comment if competing_payload&.key?('comment')
+      registration.administrative_notes = organizer_comment if competing_payload&.key?('organizer_comment')
+
+      if competing_payload&.key?('event_ids')
+        desired_events = competing_payload['event_ids']
+
+        competition_events_lookup = registration.competition.competition_events.where(event_id: desired_events).index_by(&:event_id)
+        competition_events = desired_events.map { competition_events_lookup[it] }
+
+        rce_build_params = competition_events.map { { competition_event: it } }
+        registration.registration_competition_events = registration.registration_competition_events.build(rce_build_params)
+      end
```

Comment:
This is just a fancy way of saying "try and load the `competition_events` based on WCA Event ID, or fall back to `nil` if the competition doesn't list it"